docs/user-guide/garbage-collector.md

```diff
@@ -0,0 +1,21 @@
+---
+---
+
+* TOC
+{:toc}
+
+## WARNING: Garbage Collector is an alpha feature, use it at your own risk!
+
+### What does the garbage collector do
+
+The garbage collector (GC) monitors the cluster and checks the `metadata.ownerReferences` field of each object. If none of the owners present in `metadata.ownerReferences` exists in the cluster, the GC will request the API server to delete the object.
+
```

Comment:
> How long does GC take?

We don't have numbers yet.

>  Is there any way to be notified when GC is complete?
No, there is no special notification mechanism yet. We have thought about this and have found solutions, but I don't want to mention it in the doc as we don't have a timeline yet.